napi_abstract.md

```diff
@@ -0,0 +1,263 @@
+
+### *napi_typeof*
+
+#### Signature
+```C
+napi_status napi_typeof(napi_env env, napi_value value, napi_valuetype* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: The JavaScript value whose type to query
+- `[out] result`: The type of the JavaScript value
+
+#### Return value
+- `napi_ok` if the API succeeded.
+- `napi_invalid_arg` if the type of `value` is not a known ECMAScript type and
+ `value` is not an External value.
+
+#### Description
+This N-API API represents behavior similar to invoking `typeof` Operator on 
+the object as defined in 
+[Section 12.5.5](https://tc39.github.io/ecma262/#sec-typeof-operator)
+of the ECMAScript Language Specification. However, it has support for 
+detecting External value. If `value` has a type that is invalid, an error is
+returned.
+
+### *napi_instanceof*
+
+#### Signature
+```C
+napi_status napi_instanceof(napi_env env, 
+                            napi_value object, 
+                            napi_value constructor, 
+                            bool* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  object`: The JavaScript value to check
+- `[in]  constructor`: The JavaScript function object of the constructor 
+function to check against
+- `[out] result`: Whether `object instanceof constructor` is true.
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This N-API API represents invoking `instanceof` Operator on the object as 
+defined in 
+[Section 12.10.4](https://tc39.github.io/ecma262/#sec-instanceofoperator)
+of the ECMAScript Language Specification.
+
+### *napi_is_array*
+
+#### Signature
+```C
+napi_status napi_is_array(napi_env env, napi_value value, bool* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: The JavaScript value to check
+- `[out] result`: Whether the given object is an array
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+#### Description
+This N-API API represents invoking `IsArray` operation on the object 
+as defined in [Section 7.2.2](https://tc39.github.io/ecma262/#sec-isarray)
+of the ECMAScript Language Specification.
+
+### *napi_is_arraybuffer*
+
+#### Signature
+```C
+napi_status napi_is_arraybuffer(napi_env env, napi_value value, bool* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: The JavaScript value to check
+- `[out] result`: Whether the given object is an ArrayBuffer
+
+#### Return value
+- `napi_ok` if the API succeeded.
+
+### *napi_is_buffer*
+
+#### Signature
+```C
+napi_status napi_is_buffer(napi_env env, napi_value value, bool* result)
+```
+
+#### Parameters
+- `[in]  env`: The environment that the API is invoked under
+- `[in]  value`: The JavaScript value to check
+- `[out] result`: Whether the given object is an N-API representation of 
```

Comment:
Isn't an N-API correct? I think its "a NAPI" but "an N-API" - IIRC the article depends on whether the first syllable makes a vowel sound, not whether the first letter is a vowel? But I'm not a native speak so I'll defer to @jasongin's command of the English language 😄